src/core/StelTexture.cpp

```diff
@@ -279,15 +286,21 @@ bool StelTexture::getDimensions(int &awidth, int &aheight)
 	return true;
 }
 
-QByteArray StelTexture::convertToGLFormat(QImage image, GLint& format, GLint& type, int& width, int& height)
+QByteArray StelTexture::convertToGLFormat(QImage image, GLint& format, GLint& type, int decimate, int& width, int& height)
 {
 	QByteArray ret;
 	const auto glInfo = StelMainView::getInstance().getGLInformation();
-	width = std::min(image.width(), glInfo.maxTextureSize);
-	height = std::min(image.height(), glInfo.maxTextureSize);
+	width = std::min(image.width()/decimate, glInfo.maxTextureSize);
+	height = std::min(image.height()/decimate, glInfo.maxTextureSize);
+
+#ifndef NDEBUG
+	if (decimate>1)
+		qDebug() << "decimated texture width: " << image.width() << "/" << decimate << "->" << width;
+#endif
 	if(width != image.width() || height != image.height())
 	{
-		qWarning().nospace() << "Got a texture with too large dimensions: "
+		if (decimate==1)
```

Comment:
This is not a correct condition. We should still do the print if e.g. `decimate>1`, and the decimated size is too small to fit in the limits.

Reply:
I think the warning should only be emitted if it does not fit for technical reasons. OK, there is a chance that the decimated size still does not fit the limits, but this would e.g. need a 16k texture, decimate=4 and 2k limit.